File: VB/MVVMExpenses/Views/MainView.vb
Imports System
Imports System.ComponentModel
Imports System.Windows.Forms
Imports DevExpress.Mvvm
Imports DevExpress.XtraEditors
Imports MVVMExpenses.ViewModels

Namespace MVVMExpenses
    Partial Public Class MainView
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
            Me.Opacity = 0
            If Not DesignMode Then
                InitializeNavigation()
            End If
            AddHandler ribbonControl1.Merge, AddressOf ribbonControl1_Merge
        End Sub
        Private Sub ribbonControl1_Merge(ByVal sender As Object, ByVal e As DevExpress.XtraBars.Ribbon.RibbonMergeEventArgs)
            ribbonControl1.SelectedPage = e.MergedChild.SelectedPage
        End Sub
        Private Sub InitializeNavigation()
            Dim fluentAPI = mvvmContext1.OfType(Of MyDbContextViewModel)()
            fluentAPI.BindCommand(biAccounts, Sub(x, m) x.Show(m), Function(x) x.Modules(0))
            fluentAPI.BindCommand(biCategories, Sub(x, m) x.Show(m), Function(x) x.Modules(1))
            fluentAPI.BindCommand(biTransactions, Sub(x, m) x.Show(m), Function(x) x.Modules(2))
            '
            fluentAPI.BindCommand(biLogout, Sub(x) x.Logout())
            '
            fluentAPI.WithEvent(Me, "Load").EventToCommand(Sub(x) x.OnLoaded(Nothing), Function(x) x.DefaultModule)
            fluentAPI.WithEvent(Of FormClosingEventArgs)(Me, "FormClosing").EventToCommand(Sub(x) x.OnClosing(Nothing), New Func(Of CancelEventArgs, Object)(Function(args) args))
            fluentAPI.SetTrigger(Function(x) x.State, Sub(state)
                                                          If state = AppState.Authorized Then
                                                              Opacity = 1
                                                          End If
                                                          If state = AppState.ExitQueued Then
                                                              Close()
                                                          End If
                                                      End Sub)
            Messenger.Default.Register(Of String)(Me, AddressOf OnUserNameMessage)
        End Sub
        Public Sub OnUserNameMessage(ByVal userName As String)
            If String.IsNullOrEmpty(userName) Then
                Me.Text = "Expenses Application"
            Else
                Me.Text = "Expenses Application - (" & userName & ")"
            End If
        End Sub
    End Class
End Namespace

File: CS/MVVMExpenses/Views/MainView.cs
using System;
using System.ComponentModel;
using System.Windows.Forms;
using DevExpress.Mvvm;
using DevExpress.XtraEditors;
using MVVMExpenses.Models.ViewModels;

namespace MVVMExpenses {
    public partial class MainView : XtraForm {
        public MainView() {
            InitializeComponent();
            this.Opacity = 0;
            if(!DesignMode)
                InitializeNavigation();
            ribbonControl1.Merge += ribbonControl1_Merge;
        }
        void ribbonControl1_Merge(object sender, DevExpress.XtraBars.Ribbon.RibbonMergeEventArgs e) {
            ribbonControl1.SelectedPage = e.MergedChild.SelectedPage;
        }
        void InitializeNavigation() {
            var fluentAPI = mvvmContext1.OfType<MyDbContextViewModel>();
            fluentAPI.BindCommand(biAccounts, (x, m) => x.Show(m), x => x.Modules[0]);
            fluentAPI.BindCommand(biCategories, (x, m) => x.Show(m), x => x.Modules[1]);
            fluentAPI.BindCommand(biTransactions, (x, m) => x.Show(m), x => x.Modules[2]);
            //
            fluentAPI.BindCommand(biLogout, x => x.Logout());
            //
            fluentAPI.WithEvent(this, "Load")
                .EventToCommand(x => x.OnLoaded(null), x => x.DefaultModule);
            fluentAPI.WithEvent<FormClosingEventArgs>(this, "FormClosing")
                .EventToCommand(x => x.OnClosing(null), new Func<CancelEventArgs, object>((args) => args));
            fluentAPI.SetTrigger(x => x.State, (state) =>
            {
                if(state == AppState.Authorized)
                    Opacity = 1; /*Show Main Form*/
                if(state == AppState.ExitQueued)
                    Close(); // exit the app;
            });
            Messenger.Default.Register<string>(this, OnUserNameMessage);
        }
        void OnUserNameMessage(string userName) {
            if(string.IsNullOrEmpty(userName))
                this.Text = "Expenses Application";
            else
                this.Text = "Expenses Application - (" + userName + ")";
        }
    }
}